src/plugins/video/index.js

```diff
@@ -1,68 +1,69 @@
-function video(name, deps) {
-  console.log('This is where the video plugin code would execute in the node process.');
-  //instance variables
-  this.deps = deps;
-  //hold a reference to the plugin dependencies if you are going to use them
-  this.cockpit = deps.cockpit;
-  //explicitly calling out cockpit eventemitter
-  var self = this;
-  this.deps.globalEventLoop.on('CameraRegistration', function (data) {
-    self.cockpit.emit('CameraRegistration', data);
-    //console.log('### CameraRegistration');
-  });
-  this.cameras = {};
-  function AddCamera(camera) {
-    if (self.cameras[camera.path] == undefined) {
-      self.cameras[camera.path] = [];
-    }
-    self.cameras[camera.path].push(camera);
-  }
-  this.deps.globalEventLoop.on('video-deviceRegistration', function (data) {
-    if (typeof data == 'array') {
-      data.forEach(function (camera) {
-        AddCamera(camera);
-      });
-    } else {
-      AddCamera(data);
-    }
-  });
-}
-// Start is executed after all plugins have loaded. Activate listeners here.
-video.prototype.start = function start() {
-  var self = this;  //set closure state variable for use in functions
-                    //enumerate video devices
-};
-// This is all that is required to enable settings for a plugin. They are
-// automatically made editable, saved, etc from this definition.
-video.prototype.getSettingSchema = function getSettingSchema() {
-  //from http://json-schema.org/videos.html
-  var cameraIDs = [];
-  //This should keep the CamerIDs list up to date in the schema
-  //TODO: Test, we are probabaly caching all of this so that we
-  //may need to support syncronous timing of some sort here.
-  this.deps.globalEventLoop.on('video-deviceRegistration', function (data) {
-    if (typeof data == 'array') {
-      cameraIDs = cameraIDs.concat(data);
-    } else {
-      cameraIDs.push(data.deviceid);
-    }
-  });
-  return [{
-      'title': 'Cameras & Video',
-      'category' : 'video',
-      'id': 'videosettings',
-      'type': 'object',
-      'properties': {
-        'show-stats': {
-          'title' : 'Show video player statistics',
-          'description' : 'Overlays real-time data about the video on top of the video feed. Does not affect your recorded or streaming video feed.',
-          'id': 'show-stats',
-          'type': 'boolean',
-          'default': false
+(function() 
```

Comment:
Alright, I thought it was a pattern that cockpit required for loading plugins. I originally copied it from rovpilot.js and for other plugins (servo, lights, etc). Can start undoing those as well.